### ISSUE TITLE
v0.522 carrier group

### ISSUE THREAD

**Issue by ezucemkeys** [dOpensource/dsiprouter#200]
Fresh install on deb9 from git 0.522, first login click carrier group, add, group name: test, user/password auth, username: 210, password: password, registration server 192.168.2.14, apply, resulting error attached.
![carrier_group_add](https://user-images.githubusercontent.com/51245736/61182642-39b8c080-a604-11e9-9f20-270feb800b6e.PNG)

**Comment by mackhendricks** [comment]
I will try to reproduce this issue

**Comment by mackhendricks** [comment]
Please send us your /var/log/dsiprouter.log.  This will help us understand what's happening

**Comment by ezucemkeys** [comment]
@mackhendricks @OmariKing I reverted vm snapshot to base os and attempted again today:

1. Install method on deb9 .. 
apt-get update;apt-get install -y git curl;cd /opt;git clone https://github.com/dOpensource/dsiprouter.git;cd dsiprouter;./dsiprouter.sh install -all -servernat

2. first login as admin nav to carrier group, click add, select username/pass auth radial, group name 'test', username 210, password entered

Screenshot and log attached.
![reproduce1](https://user-images.githubusercontent.com/51245736/62225303-2dfc2680-b386-11e9-95de-6441745b4287.PNG)
![reproduce2](https://user-images.githubusercontent.com/51245736/62225304-2dfc2680-b386-11e9-8960-2f49912dc19e.PNG)
[dsiprouter.log](https://github.com/dOpensource/dsiprouter/files/3452646/dsiprouter.log)

Thanks,
Matt

**Comment by ezucemkeys** [comment]
@mackhendricks @OmariKing I retested tonight using fresh install centos7. I deleted all default carrier groups and created a test ip/auth carrier group. I'm able to create a carrier group without an error, but registration failed. Checking itsp server side logs, dsiprouter is inserting a Contact: of unknown origin by default. This is the originating REGISTER message. I've also attached the full dialog as seen from itsp server side proxy log.

2020-03-02T02:04:34.699942Z:1032:INCOMING:INFO:sipx.home.mattkeys.net:SipClientUdp-11:7f2e99736700:sipxproxy:Read SIP message:
----Local Host:192.168.1.14---- Port: 5060----
----Remote Host:192.168.1.12---- Port: 5060----
REGISTER sip:192.168.1.14 SIP/2.0
Via: SIP/2.0/UDP 192.168.1.12;branch=z9hG4bKb3b7.b8f40403000000000000000000000000.0
To: <sip:205@192.168.1.14>
From: <sip:205@192.168.1.14>;tag=3393f0703fb0ccaca74109ff37de39f5-c383
CSeq: 10 REGISTER
Call-ID: 01f5d3f7440efe11-28441@127.0.0.1
Max-Forwards: 70
Content-Length: 0
Contact: <sip:8@38.124.54.244:5060>
Expires: 60

I have no idea what/where 8@38.124.54.244:5060 is.

Also note transport is being forced as UDP. Admins should be able to select tcp or udp, or to follow srv records if a fqdn was specified. 
[dsiprouter_carrier_register.log](https://github.com/dOpensource/dsiprouter/files/4273155/dsiprouter_carrier_register.log)
